### PR TITLE
Chore small fixes

### PR DESCRIPTION
**Pull request by maxnachlinger**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sudo: false
```
Comment (maxnachlinger) [inline .travis.yml:1]: So we can hook up to travis, which is free for open source projects.

File: index.js [hunk 2]
```diff
@@ -13,40 +13,41 @@ const factory = exports.factory = (struct, key) => {
     };
 };
```
Comment (maxnachlinger) [inline index.js:15]: Moved this just above where it's used. Also changed the order of the params to reflect when one would have the data in question (you'd likely have `encryptionKey` and `iv` well before before you had `buffer`) IMHO ordering them this way makes it easier for callers to do partial application as well.

File: index.js [hunk 3]
```diff
@@ -13,40 +13,41 @@ const factory = exports.factory = (struct, key) => {
...
         const iv = Buffer.from(crypto.randomBytes(ivKeySize));
 
-        stream.pipe(concat(function(buf) {
-            return resolve(Buffer.concat([iv,symmetricallyEncryptBuffer(buf, encryptionKey, iv)]).toString('base64'));
+        stream.pipe(concat((buf) => {
+            return resolve(Buffer.concat([iv, symmetricallyEncryptBuffer(encryptionKey, iv, buf)]).toString('base64'));
         }));
 
         struct.encode(stream, payload);
+        stream.once('error', reject);
```
Comment (maxnachlinger) [inline index.js:32]: Handles encoding errors

File: index.js [hunk 4]
```diff
@@ -13,40 +13,41 @@ const factory = exports.factory = (struct, key) => {
...
 
         return stream.end();
     });
 };
```
Comment (maxnachlinger) [inline index.js:37]: Move param order changes.

File: index.spec.js [hunk 5]
```diff
@@ -12,22 +12,22 @@ const RequestId = new r.VersionedStruct(r.uint8, {
 const encryptionKey = 'my test key';
 
 describe('symmetric encryption', () => {
-    const iv = new Buffer(crypto.randomBytes(16));
+    const iv = Buffer.from(crypto.randomBytes(16));
```
Comment (maxnachlinger) [inline index.spec.js:15]: `new Buffer` is deprecated

**Review by justinabrahms — APPROVED** [review]

**Pull request merged**